stdio/bignum.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * bignum.c
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Andrzej Stalke
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "bignum.h"
+
+
+static void bignum_shiftrInternal(bignum_t *bn, const size_t shift)
+{
+	bignum_data_t carry = 0, temp;
+	const bignum_data_t mask = BIGNUM_DATA_MASK >> (BIGNUM_DATA_BITS - shift);
+	ssize_t i;
+	for (i = bn->len - 1; i >= 0; --i) {
+		temp = bn->data[i];
+		if (shift == BIGNUM_DATA_BITS) {
+			bn->data[i] = 0;
+		}
+		else {
+			bn->data[i] >>= shift;
+		}
+		bn->data[i] |= carry;
+		carry = (temp & mask) << (BIGNUM_DATA_BITS - shift);
+	}
+}
+
+
+static int bignum_resize(bignum_t *bn, const size_t newSize)
+{
+	bignum_data_t *ptr = realloc(bn->data, sizeof(*bn->data) * newSize);
+	if (ptr != NULL) {
+		bn->size = newSize;
+		bn->data = ptr;
+		return 0;
+	}
+	return (-1);
```

Comment:
Is there a reason for `(-1)` instead of plain `-1`? This differs from the style of the rest of the codebase.